### DISCUSSION TITLE
How can I run a javascript in the ->afterStateUpdated() event?

### DISCUSSION THREAD

**Discussion by aytackayin** [filamentphp/filament#10597] [Help] [answered]
I want to write a JavaScript that will capture a frame of the uploaded video in the ->afterStateUpdated() of a file upload input to create a thumbnail. However, I've been unable to figure out how to execute a JavaScript function within ->afterStateUpdated(). Does anyone have expertise on this matter?

**Comment by leandrocfe** [comment, marked as answer]
```php
->afterStateUpdated(function ($livewire) {
    $livewire->js('console.log(\'xx\')');
})
```
?

**Reply by aytackayin** [reply]
I really thank you so much. I've been struggling for a week. I tried creating a Livewire component with Livewire dispatch, but it didn't work. This method saved my life. Thank you very much <3.

**Reply by JunarisAlf** [reply]
thank a lot, it work, u can do this to in action
`->action(function (Component $livewire) {
    $livewire->js('console.log("action clicked")');
})`
